### ISSUE TITLE
Execution against Oracle returns ORA-00922: missing or invalid option

### ISSUE THREAD

**Issue by twerthi** [erikbra/grate#283]
**Describe the bug**
I'm attempting to use grate against an Oracle container (gvenzl/oracle-xe) and am constantly running into ORA-00922: missing or invalid option.  The research that I've done on that error indicates that there is something wrong with the syntax of the statement I'm trying to run, however, I've manually executed it using Oracle SQL Developer as well as [DBUp](https://dbup.readthedocs.io/en/latest/) successfully.  Here's one of the statements I'm trying to run that fails

CREATE TABLE actor (
  actor_id numeric NOT NULL ,
  first_name VARCHAR(45) NOT NULL,
  last_name VARCHAR(45) NOT NULL,
  last_update DATE NOT NULL,
  CONSTRAINT pk_actor PRIMARY KEY  (actor_id)
);

**To Reproduce**
Add the following to a file called `script0001.tables.sql` in the `up` folder

CREATE TABLE actor (
  actor_id numeric NOT NULL ,
  first_name VARCHAR(45) NOT NULL,
  last_name VARCHAR(45) NOT NULL,
  last_update DATE NOT NULL,
  CONSTRAINT pk_actor PRIMARY KEY  (actor_id)
);

**Expected behavior**
The execution should successfully complete and create a table called `actor`

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows Server 2019
 - Version 1.4.0

**Additional context**

**Comment by wokket** [comment]
G'day @twerthi,

There's definitely something funny going on here, thanks for the report.

Our unit tests are run against the `oracleinanutshell/oracle-xe-11g:latest` container, and a repro on the 1.4 codebase above is giving `ORA-00911: invalid character` which matches with your syntax issue, but I'm not sure if that's something to do with my repro test, I'm not an oracle guy in any way!

Would you mind trying the scenario on your side against that container and let us know if you still get ORA-00922, ORA-00911, or something else entirely?

**Comment by twerthi** [comment]
Hey there @wokket!  Apologies for the delay, I was out on holiday :)  I switched my container to `oracleinanutshell/oracle-xe-11g:latest` and am getting the same `ORA-00911` you are.

**Comment by wokket** [comment]
Thanks mate,

I looks like it's the trailing semicolon causing the issues... I'd assume Oracle would be fine with a statement separator but the apparently not??  https://renenyffenegger.ch/notes/development/databases/Oracle/errors/ORA-00922_missing-or-invalid-option/

Again I'm not an oracle guy by any stretch so I'll defer to @erikbra for whether this is a bug (because other tools are handling this ok) or considered an input script error (because the DBMS says no).

**Comment by erikbra** [comment]
Thanks for investigating, @wokket . I would vote for this being an error in the script, in the same way that `GO` in SQL server scripts is not a valid SQL command, but a command for SQL server management studio to split batches. There is, however, some logic in `BatchSplitter` and `**Syntax.StatementSeparatorRegex`, which is supposed to handle this. I think there might be an error in the separator regex for OracleSyntax, perhaps.

I see that we have only tests for statement splitting for SQL server, here: https://github.com/erikbra/grate/tree/main/grate.unittests/Basic/Infrastructure/SqlServer/Statement_Splitting

If you are fluent with Oracle Syntax, @twerthi , do you think you can compose some similar tests to these, only for Oracle, so that we can have some tests fail, and then implement the fixes to make it work? Or, at least provide the rules that should apply for Oracle. I haven't used Oracle in ages, and don't remember all the syntax rules for it

**Comment by twerthi** [comment]
I tried running this 
```
  CREATE TABLE "SYSTEM"."TABLE1" 
   (	"COLUMN1" VARCHAR2(20 BYTE)
   ) PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 ```

I made sure there wasn't a trailing `;`, however, I still receive the `ORA-00911` running against `oracleinanutshell/oracle-xe-11g:latest`.  I too am not an Oracle expert, know just enough to get those other solutions to function, but not much more than that.  What is odd is that it would seem the statement actually worked, yet still reported the error.  Using Oracle Sql Developer, I can see table is indeed created.

**Comment by twerthi** [comment]
Scratch that, I had forgotten I had another script file in the `views` folder, that's where the error occurred this time.  The above table creation statement completed successfully.

**Comment by twerthi** [comment]
Tried running the initial statement of 
```CREATE TABLE actor (
  actor_id numeric NOT NULL ,
  first_name VARCHAR(45) NOT NULL,
  last_name VARCHAR(45) NOT NULL,
  last_update DATE NOT NULL,
  CONSTRAINT pk_actor PRIMARY KEY  (actor_id)
)```

But it's failing with 
[ServerTasks-102633.log.txt](https://github.com/erikbra/grate/files/10340431/ServerTasks-102633.log.txt)

This file is what I use for Flyway and it works, so I don't think it's syntax, but I very well could be wrong.  I realize it's not apples-to-apples in that Flyway is Java-based and Grate is .NET

[FlywayOracle.txt](https://github.com/erikbra/grate/files/10340445/FlywayOracle.txt)

Other than being a test runner, my knowledge of Oracle is quite limited.  I was attempting to add Grate as an example of doing [Oracle deployments](https://samples.octopus.app/app#/Spaces-422) with Octopus Deploy, but am unable to get past this hurdle.

**Comment by erikbra** [comment]
Hi again, @twerthi ! I really don't mind the "product placement" at all. Octopus Deploy is a wonderful tool, and I've used it a lot in the past, from the time when it was a one-man shop with Paul Stovell as the only guy on-board :)

I'd appreciate if grate were part of these examples as well, and I do think there might be grate-related problems here. The Oracle codebase hasn't got much love, and it's been ages since I used Oracle professionally myself. But, we'll get it to work anyways, won't we? :)

To make the debugging easier, since this is an "example deployment" database, would you be able to share the exact Oracle scripts that you use for the pipelines, so that I can test them here myself, and step through a bit, to see if I can pin-point the error?

**Comment by twerthi** [comment]
Certainly!  [Here](https://library.octopus.com/step-templates/ca23d18f-ab03-403d-bfb8-3ff74d3ddab3/actiontemplate-grate-database-migrations) is the code for the step template that it runs.  I've attached the package that it's using for deployment as well as what I was basing it off of, the SQL I used for Flyway.
[sakila.oracle.grate.db.1.0.23003.235231.zip](https://github.com/erikbra/grate/files/10375384/sakila.oracle.grate.db.1.0.23003.235231.zip)
[Flyway.sql.txt](https://github.com/erikbra/grate/files/10375393/Flyway.sql.txt)

At the moment, most of the grate SQL has been commented out as I was attempting to get the first statement to work properly before moving on.

**Comment by twerthi** [comment]
Hey there @erikbra!  Were you able to repro the issue?  Or was I holding it wrong?

**Comment by Dabeto83** [comment]
Hi... @twerthi, @erikbra I'm trying to use grate with oracle too and I have run into the same issue. Do you have any Idea how to solve or avoid this issue?

**Comment by Dabeto83** [comment]
Hi again  @twerthi, @erikbra I've made a change in my script I've just removed the ';' and I've added an empty line at the end of the file and it works.

[0001_CreateTables_fixed.txt](https://github.com/erikbra/grate/files/10772034/0001_CreateTables_fixed.txt)
